### PR TITLE
Improve pseudo-transaction handling (RIPD-1454, RIPD-1455)

### PR DESCRIPTION
**Pull request by bachase**
Adds additional checks and improves error messages related to pseudo-transactions.

### CODE REVIEW

File: src/test/app/PseudoTx_test.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#include <test/jtx.h>
+#include <vector>
+
+namespace ripple {
+namespace test {
+
+struct PseudoTx_test : public beast::unit_test::suite
+{
+    std::vector<std::shared_ptr<STTx>>
+    getPseudoTxs(std::uint32_t seq)
```
Comment (ximinez) [inline src/test/app/PseudoTx_test.cpp]: I am 99% sure that this doesn't need to hold `shared_ptr`s. The return value will be optimized to not make any copies even with `std::vector<STTx>`.

File: src/ripple/protocol/TxFormats.h [hunk 2]
```diff
@@ -57,6 +57,11 @@ enum TxType
     ttFEE               = 101,
 };
 
+inline bool isPseudoTx(TxType t)
+{
+    return t == ttAMENDMENT || t == ttFEE;
+}
```
Comment (bachase) [inline src/ripple/protocol/TxFormats.h]: Is it possible that [with that change](https://github.com/ximinez/rippled/commit/06de64d68c40af5d114b3a82e16170f07f518830#diff-e29a42283d218be546efb2410c71ef51R170), the caller passes in an `STObject` that lacks a `sfTransactionType` field?  That was one reason I had gone the other way, making it explicit that a transaction type is required.

**Comment by bachase** [comment]
Thanks @ximinez, I like the changes you suggested!  One question above before I fold in your changeset.

File: src/ripple/protocol/TxFormats.h (hunk 2 above)
Comment (ximinez) [inline src/ripple/protocol/TxFormats.h]: Oooooh. That's a good point.

How about
```
    auto t = tx[~sfTransactionType];
    if (!t)
        return false; // ?
    auto tt = static_cast <TxType> (*t);
    return tt == ttAMENDMENT || tt == ttFEE;
```
or
```
    if (!tx.isFieldPresent(sfTransactionType))
        return false;  // ?
    auto t = static_cast <TxType> (
        tx.getFieldU16(sfTransactionType));
    return t == ttAMENDMENT || t == ttFEE;
```

Comment (bachase) [inline src/ripple/protocol/TxFormats.h]: Works for me; I'm partial to version 1.  I'll cherry-pick and add it.

**Comment by nbougalis** [comment]
Merged as f2787dc35c8669baee29a443b2d4ea533621c891

**Pull request closed without merging**